### PR TITLE
Remove units from MOS pipeline.

### PR DESCRIPTION
**Pull request by mairanteodoro**
Resolves [RAD-911](https://jira.stsci.edu/browse/RAD-911)

Closes #

This PR addresses ...

## Tasks
- [ ] Update or add relevant `rad` tests.
- [ ] Update relevant docstrings and / or `docs/` page.
- [ ] Does this PR change any schema files?
  - [ ] Schema changes were discussed at RAD Review Board meeting.
- [ ] Does this PR change any API used downstream? (If not, label with `no-changelog-entry-needed`.)
  - [ ] Write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types).
  - [ ] Start a `romancal` regression test (https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) with this branch installed (`"git+https://github.com/<fork>/rad@<branch>"`).
  - [ ] Update relevant `roman_datamodels` utilities and tests.

<details><summary>News fragment change types:</summary>

- ``changes/<PR#>.feature.rst``: new feature
- ``changes/<PR#>.bugfix.rst``: fixes an issue
- ``changes/<PR#>.doc.rst``: documentation change
- ``changes/<PR#>.removal.rst``: deprecation or removal of public API
- ``changes/<PR#>.misc.rst``: infrastructure or miscellaneous change
</details

### CODE REVIEW

File: src/rad/resources/schemas/sky_background-1.0.0.yaml [hunk 1]
```diff
@@ -10,17 +10,8 @@ properties:
...
-          datatype:
-            enum: ["float64"]
-          unit:
-            oneOf:
-            - tag: tag:stsci.edu:asdf/unit/unit-1.*
-            - tag: tag:astropy.org:astropy/units/unit-1.*
-            enum: ["DN / s", "MJy.sr**-1"]
-      - type: "null"
+    properties:
+      datatype: float64
```
Comment (schlafly) [inline src/rad/resources/schemas/sky_background-1.0.0.yaml:14]: Somehow looking at the other cases I expected this to be type: number . This is just a scalar, right?

File: src/rad/resources/schemas/photometry-1.0.0.yaml [hunk 2]
```diff
@@ -98,13 +68,7 @@ properties:
...
       uJy /arcsec^2.
     anyOf:
-      - tag: tag:stsci.edu:asdf/unit/quantity-1.*
-        properties:
-          datatype:
-            enum: ["float64"]
-          unit:
-            tag: tag:stsci.edu:asdf/unit/unit-1.*
-            enum: ["uJy.arcsec**-2"]
+      - type: number
```
Comment (schlafly) [inline src/rad/resources/schemas/photometry-1.0.0.yaml:71]: Can we keep in the units in this case and the other scalars in this file via unit: "uJy.arcsec**-2"?  We want to preserve the information but not validate on it, as we did for the data arrays, provided that that doesn't cause some other problem?

File: src/rad/resources/schemas/wfi_mosaic-1.0.0.yaml [hunk 3]
```diff
@@ -48,28 +48,19 @@ properties:
...
-        exact_datatype: true
-        ndim: 2
-      unit:
-        tag: tag:stsci.edu:asdf/unit/unit-1.*
-        enum: ["MJy.sr**-1"]
+    tag: tag:stsci.edu:asdf/core/ndarray-1.*
+    datatype: float32
+    exact_datatype: true
+    ndim: 2
+    unit: "MJy / sr"
```
Comment (schlafly) [inline src/rad/resources/schemas/wfi_mosaic-1.0.0.yaml:55]: I notice in this case that it used to be MJy.sr**-1 and now it's MJy / sr.  It looks like we use the ".unit**power" for astropy units and DN/s for STScI units.  I guess it's fine to keep that but let's retain MJy.sr**-1 for consistency with elsewhere in rad.

File: src/rad/resources/schemas/wfi_mosaic-1.0.0.yaml [hunk 4]
```diff
@@ -48,28 +48,19 @@ properties:
...
   err:
     title: Error Data (MJy / steradian)
-    tag: tag:stsci.edu:asdf/unit/quantity-1.*
     properties:
       value:
         tag: tag:stsci.edu:asdf/core/ndarray-1.*
         datatype: float32
         exact_datatype: true
         ndim: 2
```
Comment (schlafly) [inline src/rad/resources/schemas/wfi_mosaic-1.0.0.yaml:63]: Can we keep the units in the schema but not validate on them in the same where that you did for the data array above?  Likewise for Var_poisson, var_rnoise, var_flat.

**Review by schlafly — COMMENTED** [review]
I left a few comments inline.  I think the most important thing is to try to retain the units in the schema as documentation while giving them the unit-free types so this is purely documentation and not actually triggering special behavior in the code.

File: src/rad/resources/schemas/sky_background-1.0.0.yaml (hunk 1 above)
Comment (mairanteodoro) [reply src/rad/resources/schemas/sky_background-1.0.0.yaml:14]: Done.

File: src/rad/resources/schemas/photometry-1.0.0.yaml (hunk 2 above)
Comment (mairanteodoro) [reply src/rad/resources/schemas/photometry-1.0.0.yaml:71]: Done.

File: src/rad/resources/schemas/wfi_mosaic-1.0.0.yaml (hunk 4 above)
Comment (mairanteodoro) [reply src/rad/resources/schemas/wfi_mosaic-1.0.0.yaml:63]: Done.

**Comment by mairanteodoro** [comment]
Closing this PR, since the work from here was merged into [PR 485](https://github.com/spacetelescope/rad/pull/485).

**Pull request closed without merging**